test/DynamoCoreTests/MessageLogTests.cs

```diff
@@ -27,6 +57,36 @@ public void TestWarningMessageLog()
             ProtoCore.Runtime.WarningEntry warningEntry = runtimeCore.RuntimeStatus.Warnings.ElementAt(0);
             Assert.AreEqual(ProtoCore.Runtime.WarningID.Default, warningEntry.ID);
         }
-        
+
+        [Test]
+        public void MultiThreadWarnings()
+        {
+            Assert.DoesNotThrow(() =>
+            {
+                for (int i = 0; i < 3; i++)
+                    LogWarningMessageEvents.OnLogWarningMessage($"Test{i}");
+
+                waitHandle1 = 1;// make thread1 wait until waitHandle is 0
+
+                Thread thread1 = new Thread(AddWarning);
+                thread1.Start();
+
+                while (waitHandle2 == 0 && thread1.IsAlive)// wait for thread1 to reach the Warning iteration
+                {
+                    Thread.Sleep(100);
+                }
+
+                LogWarningMessageEvents.OnLogWarningMessage($"Test3");
+
+                Interlocked.Decrement(ref waitHandle1);
```

Comment:
What happens if you don't decrement this?

Reply:
Deadlock, I guess

Reply:
This decrement call to waitHandle1 ensures that the thread1::forLOop does not finish before `Test3` warning is added